modules/pubCommonIdSystem.js

```diff
@@ -46,52 +140,78 @@ export const pubCommonIdSubmodule = {
    */
-  getId: function ({params: {create = true, pixelUrl} = {}} = {}) {
+  getId: function (config = {}) {
+    const {params: {create = true, pixelUrl} = {}} = config;
+    let newId;
     try {
       if (typeof window[PUB_COMMON_ID] === 'object') {
         // If the page includes its own pubcid module, then save a copy of id.
-        return {id: window[PUB_COMMON_ID].getId()};
+        newId = window[PUB_COMMON_ID].getId();
       }
     } catch (e) {
     }
 
-    const newId = (create && utils.hasDeviceAccess()) ? utils.generateUUID() : undefined;
+    if (!newId) newId = (create && utils.hasDeviceAccess()) ? utils.generateUUID() : undefined;
     return {
       id: newId,
-      callback: this.makeCallback(pixelUrl, newId)
+      callback: getIdCallback(newId, this.makeCallback(pixelUrl, newId), config)
     }
   },
   /**
-   * performs action to extend an id
+   * performs action to extend an id.  There are generally two ways to extend the expiration time
+   * of stored id: using pixelUrl or return the id and let main user id module write it again with
+   * the new expiration time.
+   *
+   * PixelUrl, if defined, should point back to a first party domain endpoint.  On the server
+   * side, there is either a plugin, or customized logic to read and write back the pubcid cookie.
+   * The extendId function itself should return only the callback, and not the id itself to avoid
+   * having the script-side overwriting server-side.  This applies to both pubcid and sharedid.
+   *
+   * On the other hand, if there is no pixelUrl, then the extendId should return storedId so that
+   * its expiration time is updated.  Sharedid, however, will have to be updated by this submodule
+   * separately.
+   *
    * @function
    * @param {SubmoduleParams} [config]
    * @param {Object} storedId existing id
    * @returns {IdResponse|undefined}
    */
-  extendId: function({params: {extend = false, pixelUrl} = {}} = {}, storedId) {
-    try {
-      if (typeof window[PUB_COMMON_ID] === 'object') {
-        // If the page includes its onw pubcid module, then there is nothing to do.
-        return;
-      }
-    } catch (e) {
-    }
+  extendId: function(config = {}, storedId) {
+    const {params: {extend = false, pixelUrl} = {}} = config;
 
     if (extend) {
-      // When extending, only one of response fields is needed
-      const callback = this.makeCallback(pixelUrl, storedId);
-      return callback ? {callback: callback} : {id: storedId};
+      try {
+        if (typeof window[PUB_COMMON_ID] === 'object') {
+          // If the page includes its own pubcid module, then there is nothing to do
+          // except to update sharedid
+          storeData(config, readData(config));
+          return;
+        }
+      } catch (e) {
+      }
+
+      if (pixelUrl) {
+        const callback = this.makeCallback(pixelUrl, storedId);
+        return {callback: callback};
+      } else {
+        storeData(config, readData(config));
```

Comment:
Is the intention of storeData(config, readData(config)); is to extent the expiry of the cookie? I ask this as I see the key  (config.storage.name + SHAREDID_SUFFIX ) same in readData and storeData.

Reply:
This keeps the stored value of pubcid as-is, and stores sharedid separately.  Storage of pubcid still goes thru userid modules, while this module now manages its own copy of sharedid.

Reply:
I missed the question completely, didn't I.  Yes, it's just to move the expiry time back.